packages/integration/src/bitbucketCloud/config.ts

```diff
@@ -61,8 +61,8 @@ export function readBitbucketCloudIntegrationConfig(
   const apiBaseUrl = BITBUCKET_CLOUD_API_BASE_URL;
   // If config is provided, we assume authenticated access is desired
   // (as the anonymous one is provided by default).
-  const username = config.getString('username');
-  const appPassword = config.getString('appPassword');
+  const username = config.getOptionalString('username');
```

Comment:
These fields are not optional, see https://github.com/backstage/backstage/blob/master/packages/integration/config.d.ts#L71..L80

The comment above these lines states that it expects it authenticated if the config is given.

The same fields at `BitbucketCloudIntegrationConfig` are optional though (in case of absent `bitbucketCloud` config.

only 
```
integrations:
  bitbucketCloud: {}
```
would fail but is also invalid as of `config.d.ts`

Reply:
In the case where both the suggested env variables BITBUCKET_APP_PASSWORD and BITBUCKET_USER are not present but the integration config is defined, do we want the app to fail to startup?

Reply:
Where are those end variables come from?

Reply:
> It is also possible to supply configuration through environment variables, for example `APP_CONFIG_app_baseUrl=https://staging.example.com`. 

However, as we have an array at `integrations.{name}` it is not working so well for that case.

Reply:
Hey, yes I'm referring to the [new integration config](https://github.com/backstage/backstage/blob/2ca48490106ad50510e52b11627f414b22e0c4ba/docs/integrations/bitbucketCloud/locations.md). 

My point is that if someone adds that config but the env vars referred to are not present, the whole backend crashes. As this is not a critical bit of config for the app, that doesn't seem like the desired outcome. My change allows the integration to fail gracefully - it will just 403 or 404 I guess when hitting the Bitbucket api. Does that sound ok to you?

Reply:
```yaml
integrations:
  bitbucketCloud:
    - username: ${BITBUCKET_CLOUD_USERNAME}
      appPassword: ${BITBUCKET_CLOUD_PASSWORD}
```

Ah, I see. If the env var is not present this will result in `undefined` as value (instead of e.g. an empty string), right?

In that case, I would let it fail though. You've expressed that you want an authenticated integration for Bitbucket Cloud, but you forgot to add the required env vars at the environment.

And if you want this config block to be optional, you could load it only for the environment where you need it. E.g., via a separate app-config file like
```
CMD ["node", "packages/backend", "--config", "app-config.yaml", "--config", "app-config.prod.yaml"]
```
(assuming, you only want to have it at an env you call "prod" -- name is flexible).

Reply:
> My change allows the integration to fail gracefully - it will just 403 or 404 I guess when hitting the Bitbucket api. Does that sound ok to you?

Personally, I think it should not. Then you will wonder why you get 403s even though you added the credentials until you notice that you forgot to set the env var in that environment.

I would rather let it fail early.

However, maybe the maintainers have another point of view here. @jhaals ?

Reply:
I share the same opinion as @pjungermann. In general we want to fail fast rather than starting the backend up and later realising in that we're missing important configuration.

However in reality we do not mandate a token for github or gitlab currently, the primary reason to my knowledge is that they support anonymous access. Is that something that bitbucket support? In that case it does it might be a reason to relax the config requirement. Bitbucket is however not in the default config so a user adding it to the config section is intentional perhaps the configuration should be required.

Isn't a quick solution to failing "gracefully" to only include bitbucket in the production config if that's the only place where it matters as suggested by @pjungermann above?

Reply:
@jhaals `bibucketCloud` is part of the default one with anonymous access, `bitbuckerServer` is not. For the anonymous access though, no integration config needs to be added. You only need any if you want to provide credentials.

```yaml
integrations: {}
# or absent
```
includes anonymous access for AwsS3, Azure(DevOps), GitHub, GitLab and Bitbucket Cloud; no anonymous access for Bitbucket Server and Gerrit.

Same for 
```yaml
integrations:
  bitbucketCloud: []
```

And as soon as you add an entry, `config.d.ts` requires a `username` and `appPassword` (which are the only properties as the others are non-configurable) like
```yaml
integrations:
  bitbucketCloud:
    - username: my-username
      appPassword: my-password
```

and of course, you can use env vars instead of hardcoded values
```yaml
integrations:
  bitbucketCloud:
    - username: ${BITBUCKET_CLOUD_USERNAME}
      appPassword: ${BITBUCKET_CLOUD_PASSWORD}
```
which seems not to fail fast on the absence of used env vars but later at setting up the `BitbucketCloudIntegration`.

Reply:
@pjungermann thanks for the additional context.
I discussed this with the other maintainers and we concluded that it would be a poor experience having it fail later or somehow convert to anonymous access if you have the config section but misconfigure/forget the environment variables in this case. Given the and context so far I don't it's a good idea to proceed with this change.

Reply:
Ok thanks both